File: linked_lists/lc328.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:

        if head == None or head.next == None:
            return head

        oddHead = head
        evenHead = head.next

        oddCur = oddHead
        evenCur = evenHead

        while oddCur.next is not None and evenCur.next is not None:

            oddCur.next = evenCur.next
            oddCur = oddCur.next

            evenCur.next = oddCur.next
            evenCur = evenCur.next

        oddCur.next = evenHead

        return oddHead

File: phase 2/01- LinkedList/lc328.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var oddEvenList = function(head) {
    
    if(head == null || head.next == null){return head}
    let oddHead = head
    let evenHead = head.next

    let oddCur = oddHead
    let evenCur = evenHead

    while (oddCur.next != null && evenCur.next != null){
        oddCur.next = evenCur.next
        oddCur = oddCur.next

        evenCur.next = oddCur.next
        evenCur = evenCur.next
    }

    oddCur.next = evenHead

    return oddHead


};